### PR TITLE
identify pointer type in df::world_site_realization

### PR DESCRIPTION
**Pull request by ragundo**
Identified using IDA

### CODE REVIEW

File: df.world-site.xml [hunk 1]
```diff
@@ -431,7 +431,14 @@
             <static-array count='51' pointer-type='site_realization_crossroads'/>
         </static-array>
         <static-array name='river_map' count='51'>
-            <static-array count='51' type-name='pointer'/>
+            <pointer>
+                <static-array name='start' type-name='int32_t' count='48'/>
+                <static-array name='end'   type-name='int32_t' count='48'/>
+                <int32_t name="unk_180"/>
+                <int32_t name="unk_184"/>
+                <int32_t name="start_array_limits"/>
```
Comment (lethosor) [inline df.world-site.xml]: What's this? It's kind of unclear from the name, so maybe an explanation in a comment would be helpful.

Comment (ragundo) [inline df.world-site.xml]: Yes, you are right

This is unknown struct used by DF to draw the rivers of a site when it's exported as a bitmap.

start array tells the start pixel where a river must be draw until its corresponding value in end array.

These pixels, of course are local to the subdivision that river_map[][] does of the 816 x 816 pixels that a image of a world site is.

Comment (lethosor) [inline df.world-site.xml]: I mean the `start_array_limits` field

File: df.world-site.xml [hunk 2]
```diff
@@ -431,14 +431,16 @@
...
         <static-array name='river_map' count='51'>
-            <pointer>
-                <static-array name='start' type-name='int32_t' count='48'/>
-                <static-array name='end'   type-name='int32_t' count='48'/>
-                <int32_t name="unk_180"/>
-                <int32_t name="unk_184"/>
-                <int32_t name="start_array_limits"/>
-                <int32_t name="end_array_limits"/>
-            </pointer>
+            <static-array count='51'>
```
Comment (lethosor) [inline df.world-site.xml]: Doesn't this generate a 51x51 array? I'm pretty sure that ends up being way larger than what the structure was before, so are you sure this works?

Comment (ragundo) [inline df.world-site.xml]: world_site is 816 x 816 pixels
816/51 = 16 so each entry in the map corresponds to a 16x16 area.
start is a 48 int array = 48 / 3 local pixel per pixel embark = 16, which is the previous area.

I don't have time now but tomorrow I'll submit some images to prove it

Comment (ragundo) [inline df.world-site.xml]: Also river_map is a double dimension array of pointers to the proposal unknown structure.
Previously was one array of void pointers. I only changed the type of the pointer.

[51]x[51]x4 = 10404, so there's no size difference (4 = sizeof pointer)

**Pull request merged**